test/it/uk/gov/hmrc/TestUserSpec.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it.uk.gov.hmrc
+
+import java.util.concurrent.TimeUnit
+
+import org.apache.http.HttpStatus._
+import org.scalatest._
+import play.api.Mode
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.json.Json
+import play.api.test.TestServer
+import uk.gov.hmrc.mongo.MongoConnector
+import uk.gov.hmrc.testuser.repository.TestUserMongoRepository
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import uk.gov.hmrc.testuser.models.JsonFormatters._
+
+import scala.concurrent.Await._
+import scala.concurrent.duration.Duration
+import scalaj.http.{Http, HttpResponse}
+import org.mindrot.jbcrypt.{BCrypt => BCryptUtils}
+import uk.gov.hmrc.testuser.models._
+
+class TestUserSpec extends FeatureSpec with Matchers
+with GivenWhenThen with BeforeAndAfterEach with BeforeAndAfterAll {
+
+  val port = 9999
+  val timeout = Duration(5, TimeUnit.SECONDS)
+  val serviceUrl = s"http://localhost:$port"
+  var server: TestServer = _
+
+  private def authenticate(usr: String, pwd: String): HttpResponse[String] = {
+    val payload = Json.parse(s"""{ "username": "$usr", "password" :"$pwd" }""").toString
+
+    Http(s"$serviceUrl/authenticate")
+      .headers("Content-type" -> "application/json")
+      .postData(payload).asString
+  }
+
+  feature("Create a test user") {
+
+    scenario("Create an individual") {
+
+      When("I request the creation of an individual")
+      val createdResponse = Http(s"$serviceUrl/individual").postForm.asString
+
+      Then("The response contains the details of the individual created")
+      createdResponse.code shouldBe SC_CREATED
+      val individualCreated = Json.parse(createdResponse.body).as[CreateTestIndividualResponse]
+
+      And("The individual is stored in Mongo with hashed password")
+      val individualFromMongo = result(mongoRepository.fetchByUsername(individualCreated.username), timeout).get.asInstanceOf[TestIndividual]
+      val expectedIndividualCreated = CreateTestIndividualResponse.from(individualFromMongo.copy(password = individualCreated.password))
+      individualCreated shouldBe expectedIndividualCreated
+      validatePassword(individualCreated.password, individualFromMongo.password) shouldBe true
+
+      And("If I login with the individual's credentials")
```

Comment:
Can we move this part to another feature as it is a very long test?
feature("Authenticate User") {
scenario("Valid credentials") {
...
}
scenario("Invalid credentials") {
...
}
}